src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java

```diff
@@ -94,6 +95,13 @@ public Cancelable autoAnalyzeFiles(Collection<VirtualFile> files, TriggerType tr
       return null;
     }
     var callback = new UpdateOnTheFlyFindingsCallable(onTheFlyFindingsHolder);
+
+    var lastAnalysisDate = getGlobalSettings().getLastAutoAnalysisDate();
+
+    if (lastAnalysisDate == 0) {
+      getGlobalSettings().setLastAutoAnalysisDate(Instant.now().toEpochMilli());
+    }
```

Comment:
Could we find a way to not modify the AnalysisSubmitter and rely only on the event?